gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/graph.js

```diff
@@ -46,9 +46,10 @@ class Graph {
 }
 
 class Element {
-  constructor(id, label) {
+  constructor(id, label, properties = undefined) {
```

Comment:
This breaks a bit with what `gremlin-python` does, but wouldn't it be better for this to just do `properties = []` here? Since undefined isn't the same as null it makes the assignment to `this.properties` below a little cleaner.

Reply:
I might have to take back what I said. The check needs to stay there if we are trying to prevent users from ever assigning null or undefined to properties. So I guess it depends on what should be done if a user explicitly tries to do something like `new Vertex("a", "b", null)`

Reply:
I think the undefined case is fine with either default but yea, the `null` check has to be present or else js wraps it into `[null]`. I might leave it as `[]` for clarity.